Riot/ViewController/SettingsViewController.m

```diff
@@ -3430,6 +3437,87 @@ - (void)deleteKeyExportFile
+
+            NSString *newTheme;
+            if (action == autoAction)
+            {
+                newTheme = @"auto";
+            }
+            else  if (action == lightAction)
+            {
+                newTheme = @"light";
+            }
+            else if (action == darkAction)
+            {
+                newTheme = @"dark";
+            }
+
+            NSString *theme = [[NSUserDefaults standardUserDefaults] stringForKey:@"userInterfaceTheme"];
+            if (newTheme && ![newTheme isEqualToString:theme])
+            {
+                // Clear fake Riot Avatars based on the previous theme.
+                [AvatarGenerator clear];
+
+                // The user wants to select this theme
+                [[NSUserDefaults standardUserDefaults] setObject:newTheme forKey:@"userInterfaceTheme"];
+                [[NSUserDefaults standardUserDefaults] synchronize];
+
+                [self.tableView reloadData];
+            }
+        }
+    };
+
+    if (@available(iOS 11.0, *))
+    {
+        // Show "auto" only from iOS 11
+        autoAction = [UIAlertAction actionWithTitle:NSLocalizedStringFromTable(@"settings_ui_theme_auto", @"Vector", nil)
+                                              style:UIAlertActionStyleDefault
+                                            handler:actionBlock];
+
+        // Explain what is "auto"
+        themePickerMessage = NSLocalizedStringFromTable(@"settings_ui_theme_picker_message", @"Vector", nil);
+    }
+
+    lightAction = [UIAlertAction actionWithTitle:NSLocalizedStringFromTable(@"settings_ui_theme_light", @"Vector", nil)
+                                          style:UIAlertActionStyleDefault
+                                        handler:actionBlock];
+
+    darkAction = [UIAlertAction actionWithTitle:NSLocalizedStringFromTable(@"settings_ui_theme_dark", @"Vector", nil)
+                                           style:UIAlertActionStyleDefault
+                                         handler:actionBlock];
+
+    // Ask the user the kind of the call: voice or video?
+    UIAlertController *themePicker = [UIAlertController alertControllerWithTitle:NSLocalizedStringFromTable(@"settings_ui_theme_picker_title", @"Vector", nil)
+                                                                         message:themePickerMessage
+                                                                  preferredStyle:UIAlertControllerStyleActionSheet];
+
+    if (autoAction)
+    {
+        [themePicker addAction:autoAction];
+    }
+    [themePicker addAction:lightAction];
+    [themePicker addAction:darkAction];
+
+    // Cancel button
+    [themePicker addAction:[UIAlertAction actionWithTitle:[NSBundle mxk_localizedStringForKey:@"cancel"]
+                                                        style:UIAlertActionStyleDefault
+                                                      handler:nil]];
+
+    [self presentViewController:themePicker animated:YES completion:nil];
```

Comment:
You to define [themePicker popoverPresentationController].sourceView  and sourceRect